pkg/storage/migrate/registry.go

```diff
@@ -0,0 +1,131 @@
+package migrate
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/pressly/goose/v3"
+)
+
+type migrationFunc = func(ctx context.Context, tx *sql.Tx) error
+
+type Migration struct {
+	Version  int64
+	Forward  migrationFunc
+	Backward migrationFunc
+}
+
+type Registry struct {
+	dialect goose.Dialect
+
+	Migrations map[int64]*Migration
+}
+
+func (r *Registry) Register(m *Migration) {
```

Comment:
Mind if we rename this signature to `MustRegister` to follow suite with the rest of our codebase around functions that will panic if a pre-condition is not met?